cmake/Modules/legate_core_options.cmake

```diff
@@ -0,0 +1,125 @@
+# limitations under the License.
+#=============================================================================
+
+option(BUILD_SHARED_LIBS "Build legate.core shared libraries" ON)
+
+function(set_or_default var_name var_env)
+  list(LENGTH ARGN num_extra_args)
+  if(num_extra_args GREATER 0)
+    list(GET ARGN 0 var_default)
+  endif()
+  if(DEFINED ${var_name})
+    message(VERBOSE "legate.core: ${var_name}=${${var_name}}")
+  elseif(DEFINED ENV{${var_env}})
+    set(${var_name} $ENV{${var_env}} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=$ENV{${var_env}} (from envvar '${var_env}')")
+  elseif(DEFINED var_default)
+    set(${var_name} ${var_default} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=${var_default} (from default value)")
+  else()
+    message(VERBOSE "legate.core: not setting ${var_name}")
+  endif()
+endfunction()
+
+# Initialize these vars from the CLI, then fallback to an envvar or a default value.
+set_or_default(Legion_SPY USE_SPY OFF)
+set_or_default(Legion_USE_LLVM USE_LLVM OFF)
+set_or_default(Legion_USE_CUDA USE_CUDA OFF)
+set_or_default(Legion_USE_HDF5 USE_HDF OFF)
+set_or_default(Legion_USE_GASNet USE_GASNET OFF)
+set_or_default(Legion_USE_OpenMP USE_OPENMP OFF)
+set_or_default(Legion_BOUNDS_CHECKS CHECK_BOUNDS OFF)
+
+option(Legion_SPY "Enable detailed logging for Legion Spy" OFF)
+option(Legion_USE_LLVM "Use LLVM JIT operations" OFF)
+option(Legion_USE_HDF5 "Enable support for HDF5" OFF)
+option(Legion_USE_CUDA "Enable Legion support for the CUDA runtime" OFF)
+option(Legion_USE_GASNet "Enable the distributed GASNet backend" OFF)
+option(Legion_USE_OpenMP "Use OpenMP" OFF)
+option(Legion_USE_Python "Use Python" OFF)
+option(Legion_BOUNDS_CHECKS "Enable bounds checking in Legion accessors" OFF)
+
+if(Legion_USE_GASNet)
+  set_or_default(GASNet_ROOT_DIR GASNET)
+  set_or_default(GASNet_CONDUIT CONDUIT "udp")
+
+  if(Legion_USE_GASNet AND (NOT GASNet_ROOT_DIR))
+    option(Legion_EMBED_GASNet "Embed a custom GASNet build into Legion" ON)
+  endif()
+endif()
+
+set_or_default(Legion_MAX_DIM LEGION_MAX_DIM 4)
+
+# Check the max dimensions
+if((Legion_MAX_DIM LESS 1) OR (Legion_MAX_DIM GREATER 9))
+  message(FATAL_ERROR "The maximum number of Legate dimensions must be between 1 and 9 inclusive")
+endif()
+
+set_or_default(Legion_MAX_FIELDS LEGION_MAX_FIELDS 256)
+
+# Check that max fields is between 32 and 4096 and is a power of 2
+if(NOT Legion_MAX_FIELDS MATCHES "^(32|64|128|256|512|1024|2048|4096)$")
+  message(FATAL_ERROR "The maximum number of Legate fields must be a power of 2 between 32 and 4096 inclusive")
+endif()
+
+option(legate_core_STATIC_CUDA_RUNTIME "Statically link the cuda runtime library" OFF)
+option(legate_core_EXCLUDE_LEGION_FROM_ALL "Exclude Legion targets from legate.core's 'all' target" OFF)
+
+set_or_default(NCCL_DIR NCCL_PATH)
+set_or_default(Thrust_DIR THRUST_PATH)
+set_or_default(CUDA_TOOLKIT_ROOT_DIR CUDA)
+set_or_default(CMAKE_CUDA_ARCHITECTURES GPU_ARCH ALL)
```

Comment:
Originally I defaulted to "ALL" so it doesn't fail on CPU-only runners (i.e. in CI) that would fail the native arch detection, but I believe `rapids-cmake` now detects this case and reverts to building all archs as a fallback. I'm fine if we set this to NATIVE by default.